kibbeh/src/pages/download.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from "react";
+import { useTypeSafeTranslation } from "../shared-hooks/useTypeSafeTranslation";
+import { Button } from "../ui/Button";
+
+const links: any = {
+  "Mac OS":
+    "https://github.com/benawad/dogehouse/releases/download/{{tag}}/DogeHouse-{{v}}.dmg",
+  Windows:
+    "https://github.com/benawad/dogehouse/releases/download/{{tag}}/DogeHouse-Setup-{{v}}.exe",
+  Linux:
+    "https://github.com/benawad/dogehouse/releases/download/{{tag}}/DogeHouse-{{v}}.AppImage",
+};
+
+export default function Download() {
+  const [downloadFailed, setDownloadFailed] = useState(false);
+  const { t } = useTypeSafeTranslation();
+  useEffect(() => {
+    let os = getOS();
+    if (!os.isPhone) {
+      let res = false;
+      var xmlHttp = new XMLHttpRequest();
+      xmlHttp.open(
+        "GET",
+        "https://api.github.com/repos/benawad/dogehouse/releases/latest"
+      );
+      xmlHttp.send(null);
+      xmlHttp.onreadystatechange = () => {
+        if (xmlHttp.responseText && !res) {
+          res = true;
+          let data = JSON.parse(xmlHttp.responseText);
+          let tag = data.tag_name;
+          if (tag) {
+            let v = tag.replace("v", "");
+            let link = links[os.os].replace("{{tag}}", tag).replace("{{v}}", v);
+            window.location.href = link;
+          } else {
+            setDownloadFailed(true);
+          }
+        }
+      };
+    } else {
+      setDownloadFailed(true);
+    }
+  }, []);
+
+  return (
+    <div className="flex w-full h-full flex-col items-center justify-center p-8">
+      <h4 className="text-primary-100 mb-4">
+        {downloadFailed
+          ? t("pages.download.failed")
+          : t("pages.download.starting")}
+      </h4>
+      {downloadFailed ? (
+        <Button
+          onClick={() => {
+            window.location.href =
+              "https://github.com/benawad/dogehouse/releases/latest";
+          }}
+        >
+          {t("pages.download.visit_gh")}
+        </Button>
+      ) : null}
+    </div>
+  );
+}
+
+function getOS() {
+  let isWindows = false;
+  let isMac = false;
+  let isLinux = false;
+  let isPhone = false;
+
+  const userAgent = window.navigator.userAgent;
+  const platform = window.navigator.platform;
+  const macosPlatforms = ["Macintosh", "MacIntel", "MacPPC", "Mac68K"];
+  const windowsPlatforms = ["Win32", "Win64", "Windows", "WinCE"];
+  const iosPlatforms = ["iPhone", "iPad", "iPod"];
+
+  let os = "Windows";
```

Comment:
nit: Might be useful to set the os to "" instead of "Windows" in the case where a platform name doesn't exist in these arrays

Reply:
i did set it to `""` before but typescript didnt like it. that's why i set it to windows now